akka-osgi/src/main/scala/akka/osgi/BundleDelegatingClassLoader.scala

```diff
@@ -76,20 +76,20 @@ class BundleDelegatingClassLoader(bundle: Bundle, fallBackClassLoader: ClassLoad
           process(processed, rest)
         } else {
           val wiring = b.adapt(classOf[BundleWiring])
-          val direct: Set[Bundle] =
-            if (wiring == null) Set.empty
+          val direct: Seq[Bundle] =
+            if (wiring == null) Seq.empty
             else {
-              val requiredWires: List[BundleWire] =
-                wiring.getRequiredWires(BundleRevision.PACKAGE_NAMESPACE).asScala.toList
+              val requiredWires: Seq[BundleWire] =
+                wiring.getRequiredWires(BundleRevision.PACKAGE_NAMESPACE).asScala.toSeq
               requiredWires.flatMap {
                 wire ⇒ Option(wire.getProviderWiring) map { _.getBundle }
-              }.toSet
+              }
             }
-          process(processed + b, rest ++ (direct -- processed))
+          process(Seq(b) ++ processed, rest ++ direct.diff(processed))
```

Comment:
Since `processed` is only prepended to, why not just call it a `List` and use `::`?